File: javascript-closure-angular-client/API/Client/Vehicle.js
goog.provide('API.Client.Vehicle');

/**
 * @record
 */
API.Client.Vehicle = function() {}

/**
 * - 1 # Sales by car dealership - 2 # Direct billing - 3 # Direct sales - 0 # Other 
 * @type {!number}
 * @export
 */
API.Client.Vehicle.prototype.tpOp;

/**
 * @type {!string}
 * @export
 */
API.Client.Vehicle.prototype.chassisNumber;

/**
 * color code used by factory
 * @type {!string}
 * @export
 */
API.Client.Vehicle.prototype.colorCode;

/**
 * color name
 * @type {!string}
 * @export
 */
API.Client.Vehicle.prototype.colorName;

/**
 * Power using CV unid
 * @type {!string}
 * @export
 */
API.Client.Vehicle.prototype.cvPower;

/**
 * engine size in cubic centimeters
 * @type {!string}
 * @export
 */
API.Client.Vehicle.prototype.cylinderVolumCC;

/**
 * net weight
 * @type {!string}
 * @export
 */
API.Client.Vehicle.prototype.netWeight;

/**
 * gross weight
 * @type {!string}
 * @export
 */
API.Client.Vehicle.prototype.grossWeight;

/**
 * serie
 * @type {!string}
 * @export
 */
API.Client.Vehicle.prototype.serialNumber;

/**
 * - '01' # Alcohol - '02' # Gas - '03' # Diesel - '16' # Alcohol/Gas - '17' # Gas/Alcohol/GNV - '18' # Gas/Electric 
 * @type {!string}
 * @export
 */
API.Client.Vehicle.prototype.fuelType;

/**
 * engine number
 * @type {!string}
 * @export
 */
API.Client.Vehicle.prototype.engineNumber;

/**
 * traction capacity, tonne
 * @type {!string}
 * @export
 */
API.Client.Vehicle.prototype.cmt;

/**
 * length between axis
 * @type {!string}
 * @export
 */
API.Client.Vehicle.prototype.lengthBetweenAxis;

/**
 * model year
 * @type {!string}
 * @export
 */
API.Client.Vehicle.prototype.modelYear;

/**
 * year of factory
 * @type {!string}
 * @export
 */
API.Client.Vehicle.prototype.manufactoryYear;

/**
 * paynt type
 * @type {!string}
 * @export
 */
API.Client.Vehicle.prototype.paintType;

/**
 * @type {!string}
 * @export
 */
API.Client.Vehicle.prototype.vehicleRENAVAMType;

/**
 * @type {!string}
 * @export
 */
API.Client.Vehicle.prototype.specieRENAVAMType;

/**
 * @type {!string}
 * @export
 */
API.Client.Vehicle.prototype.modelRENAVAMCode;

/**
 * - '01' # YELLOW - '02' # BLUE - '03' # BEIGE - '04' # WHITE - '05' # GRAY - '06' # GOLDEN - '07' # GRENA - '08' # ORANGE - '09' # BROWN - '10' # SILVER - '11' # BLACK - '12' # PINK - '13' # PURPLE - '14' # GREEN - '15' # RED - '16' # FANTASY 
 * @type {!string}
 * @export
 */
API.Client.Vehicle.prototype.colorDENATRANCode;

/**
 * chassis condition original or modified (remarcado)
 * @type {!string}
 * @export
 */
API.Client.Vehicle.prototype.vin;

/**
 * - '1' # Finished(Done) - '2' # Not finished - '3' # Semi-finished 
 * @type {!string}
 * @export
 */
API.Client.Vehicle.prototype.vehicleManufactoryStatus;

/**
 * max number of sit occupants
 * @type {!string}
 * @export
 */
API.Client.Vehicle.prototype.maxOccupantsQuantity;

/**
 * Lien/restriction type - '0' # No one; - '1' # Alienação Fiduciária; - '2' # Arrendamento Mercantil; - '3' # Reserva de Domínio; - '4' # Penhor de Veículos; - '9' # Other. 
 * @type {!string}
 * @export
 */
API.Client.Vehicle.prototype.restrictionType;

/** @enum {string} */
API.Client.Vehicle.FuelTypeEnum = { 
  01: '01',
  02: '02',
  03: '03',
  16: '16',
  17: '17',
  18: '18',
}
/** @enum {string} */
API.Client.Vehicle.ColorDENATRANCodeEnum = { 
  01: '01',
  02: '02',
  03: '03',
  04: '04',
  05: '05',
  06: '06',
  07: '07',
  08: '08',
  09: '09',
  10: '10',
  11: '11',
  12: '12',
  13: '13',
  14: '14',
  15: '15',
  16: '16',
}
/** @enum {string} */
API.Client.Vehicle.VinEnum = { 
  R: 'R',
  N: 'N',
}
/** @enum {string} */
API.Client.Vehicle.VehicleManufactoryStatusEnum = { 
  1: '1',
  2: '2',
  3: '3',
}
/** @enum {string} */
API.Client.Vehicle.RestrictionTypeEnum = { 
  0: '0',
  1: '1',
  2: '2',
  3: '3',
  4: '4',
  9: '9',
}

File: async-scala-client/src/main/scala/io/swagger/client/model/Vehicle.scala
package io.swagger.client.model

import org.joda.time.DateTime
import java.util.UUID


case class Vehicle (
  tpOp: Integer,  // - 1 # Sales by car dealership - 2 # Direct billing - 3 # Direct sales - 0 # Other 
chassisNumber: String,
colorCode: String,  // color code used by factory
colorName: String,  // color name
cvPower: String,  // Power using CV unid
cylinderVolumCC: String,  // engine size in cubic centimeters
netWeight: String,  // net weight
grossWeight: String,  // gross weight
serialNumber: String,  // serie
fuelType: String,  // - &#39;01&#39; # Alcohol - &#39;02&#39; # Gas - &#39;03&#39; # Diesel - &#39;16&#39; # Alcohol/Gas - &#39;17&#39; # Gas/Alcohol/GNV - &#39;18&#39; # Gas/Electric 
engineNumber: String,  // engine number
cmt: String,  // traction capacity, tonne
lengthBetweenAxis: String,  // length between axis
modelYear: String,  // model year
manufactoryYear: String,  // year of factory
paintType: String,  // paynt type
vehicleRENAVAMType: String,
specieRENAVAMType: String,
modelRENAVAMCode: String,
colorDENATRANCode: String,  // - &#39;01&#39; # YELLOW - &#39;02&#39; # BLUE - &#39;03&#39; # BEIGE - &#39;04&#39; # WHITE - &#39;05&#39; # GRAY - &#39;06&#39; # GOLDEN - &#39;07&#39; # GRENA - &#39;08&#39; # ORANGE - &#39;09&#39; # BROWN - &#39;10&#39; # SILVER - &#39;11&#39; # BLACK - &#39;12&#39; # PINK - &#39;13&#39; # PURPLE - &#39;14&#39; # GREEN - &#39;15&#39; # RED - &#39;16&#39; # FANTASY 
vin: String,  // chassis condition original or modified (remarcado)
vehicleManufactoryStatus: String,  // - &#39;1&#39; # Finished(Done) - &#39;2&#39; # Not finished - &#39;3&#39; # Semi-finished 
maxOccupantsQuantity: String,  // max number of sit occupants
restrictionType: String  // Lien/restriction type - &#39;0&#39; # No one; - &#39;1&#39; # Alienação Fiduciária; - &#39;2&#39; # Arrendamento Mercantil; - &#39;3&#39; # Reserva de Domínio; - &#39;4&#39; # Penhor de Veículos; - &#39;9&#39; # Other. 
)
